sdk/core/azure-core/src/http/curl/curl.cpp

```diff
@@ -2235,6 +2242,45 @@ std::unique_ptr<CurlNetworkConnection> CurlConnectionPool::ExtractOrCreateCurlCo
   return std::make_unique<CurlConnection>(request, options, hostDisplayName, connectionKey);
 }
 
+Azure::Core::Http::_detail::KeepAliveOptions CurlConnection::ParseKeepAliveHeader(
+    std::string const& keepAlive)
+{
+  Azure::Core::Http::_detail::KeepAliveOptions keepAliveOptions;
+  // Parse the Keep-Alive header to determine if the connection should be kept alive.
+  // The Keep-Alive header is in the format of:
+  // Keep-Alive: timeout=5, max=1000
+  // The timeout is the number of seconds the connection is allowed to be idle before it is closed.
+  // The max is the maximum number of requests that can be made on the connection before it is
+  // closed.
+  // If the header is not present, the connection will be kept alive for the lifetime of the
+  // application.
+  std::string const timeoutKey = "timeout=";
+  std::string const maxKey = "max=";
+  auto timeoutPos = keepAlive.find(timeoutKey);
+  auto maxPos = keepAlive.find(maxKey);
+  if (timeoutPos != std::string::npos)
+  {
+    auto timeoutEnd = keepAlive.find(',', timeoutPos);
```

Comment:
This assumes that the Keep-Alive header always has the form: `keep-alive: timeout=<number>, max=<number>`

Can you be guaranteed that we will always see that format? Or is `keep-alive: timeout=<number>` or `keep-alive: max=5, timeout=27` valid?

Reply:
You might want to consider using something like `StringExtensions::split` to parse the keep alive header. Something like: 

```
std::map<std::string, std::string> keepAliveOptions;
std::vector<std::string> values = StringExtensions::Split(keepAlive, ","); 
for (const auto &s in values) {
  std::vector<std::string> options = StringExtensions::Split(s, "=");
  keepAliveOptions.emplace(options[0], options[1]);
}
std::string& timeout = keepAliveOptions["timeout"];
std::string max=keepAliveOptions["max"];
```

This is missing a TON of checks, but it should work for all the possible variations of keep-alive.

It's just a thought.

Reply:
added tests and some refactor of it